### ISSUE TITLE
PersistentVolumeClaim error

### ISSUE THREAD

**Issue by DrumSergio** [MoJo2600/pihole-kubernetes#41]
Hi. I am following this tutorial: https://kauri.io/68-selfhost-pihole-on-kubernetes-and-block-ads-and/5268e3daace249aba7db0597b47591ef/a

In "Deployment" section, point 4a, the changes made to the "values.yml" file somehow can't be installed through helm. This is the error I am facing:
```
Error: Deployment in version "v1" cannot be handled as a Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Volumes: []v1.Volume: v1.Volume.VolumeSource: PersistentVolumeClaim: v1.PersistentVolumeClaimVolumeSource.ClaimName: ReadString: expects " or n, but found t, error found in #10 byte of ...|aimName":true}},{"co|..., bigger context ...|e":"config","persistentVolumeClaim":{"claimName":true}},{"configMap":{"defaultMode":420,"name":"piho|...
```

When I perform a "kubectl get pvc -n pihole" this is the output:
```
NAME     STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
pihole   Bound    pihole   500Mi      RWO            manual         3h30m
```

PS. I checked-out the commit from March and it is working now, so latest commit is not working

**Comment by i5Js** [comment]
Hi, 
Can you share your values file?
Regards

**Comment by DrumSergio** [comment]
```
# Default values for pihole.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.

replicaCount: 2

image:
  repository: "pihole/pihole"
  tag: 4.3.2-1
  pullPolicy: IfNotPresent

serviceTCP:
  type: LoadBalancer # Configure MetalLB to used a dedicated "virtual" IP to expose the DNS server
  annotations:
    metallb.universe.tf/allow-shared-ip: pihole-svc

serviceUDP:
  type: LoadBalancer # Configure MetalLB to used a dedicated "virtual" IP to expose the DNS server
  annotations:
    metallb.universe.tf/allow-shared-ip: pihole-svc

ingress:
  enabled: false
  annotations: {}
    # kubernetes.io/ingress.class: nginx
    # kubernetes.io/tls-acme: "true"
  path: /
  hosts:
    - chart-example.local
  tls: []
  #  - secretName: chart-example-tls
  #    hosts:
  #      - chart-example.local

# Probes configuration
probes:
  # probes.liveness -- Configure the healthcheck for the ingress controller
  liveness:
    enabled: true
    initialDelaySeconds: 60
    failureThreshold: 10
    timeoutSeconds: 5
  readiness:
    enabled: true
    initialDelaySeconds: 60
    failureThreshold: 3
    timeoutSeconds: 5

resources: {}
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube. If you do want to specify resources, uncomment the following
  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
  # limits:
  #  cpu: 100m
  #  memory: 128Mi
  # requests:
  #  cpu: 100m
  #  memory: 128Mi

persistentVolumeClaim:
  # set to true to use pvc
  enabled: true # Change to true
  # set to true to use you own pvc
  existingClaim: true # Change to true, it will automatically pick the persistent volume claim named "pihole"

nodeSelector: {}

tolerations: []

affinity: {}

# Administrator password when not using an existing secret (see below)
adminPassword: "admin"

# Use an existing secret for the admin password.
admin:
  existingSecret: "pihole-secret"
  passwordKey: "password"

# extraEnvironmentVars is a list of extra enviroment variables to set for pihole to use
extraEnvVars:
  TZ: "Europe/Madrid" # Timezone
  ServerIP: 10.0.0.100
# default upstream DNS servers to use
  DNS1: "1.1.1.1"
  DNS2: "8.8.8.8"

doh:
  # set to true to enabled DNS over HTTPs via cloudflared
  enabled: false
  name: "cloudflared"
  repository: "crazymax/cloudflared"
  tag: latest
  pullPolicy: IfNotPresent

dnsmasq:
  customDnsEntries: []
  # Here you can add custom dns entries to override the
  # dns resolution with. All lines will be added to the
  # pihole dnsmasq configuration.
  # The format should be like:
  # - address=/foo.bar/192.168.178.10
  # - address=/bar.foo/192.168.178.11

  additionalHostsEntries: []
  # Dnsmasq reads the /etc/hosts file to resolve ips. You can add additional entries if you like
  # - 192.168.0.3     host4
  # - 192.168.0.4     host5


adlists: {}
  # If you want to provide blocklists, add them here.
  # - https://hosts-file.net/grm.txt
  # - https://reddestdream.github.io/Projects/MinimalHosts/etc/MinimalHostsBlocker/minimalhosts

whitelist: {}
  # If you want to provide whitelisted domains, add them here.
  # - clients4.google.com

blacklist: {}
  # If you want to have special domains blacklisted, add them here
  # - *.blackist.com

regex: {}
  # Add regular expression blacklist items
  # - (^|\.)facebook\.com$


```

**Comment by i5Js** [comment]
I don't see any gap here, it's similar and mine is working fine. do you still have the issue?

**Comment by DrumSergio** [comment]
No, I don't have the issue anymore, but that is because I am not using the latest commit. Something should be failing in the latest commit around the "persistentVolumeClaim". It is strange.

**Comment by gjeanmart** [comment]
Hey,
Lastest committer here :) 

In my recent PR which was merged, I have changed the template to be more compliant with Help chart best practices regarding persistantVolumeClaim.

Basically, instead of having `persistentVolumeClaim.existingClaim:true/false` and force people to name the claim `pihole`, this field is now expecting a string `persistentVolumeClaim.existingClaim:<name-of-my-claim>` (can be `pihole` or anything else).

Hope that makes sense.

[Here is an example with nextcloud from the official Helm repo](https://github.com/helm/charts/blob/master/stable/nextcloud/templates/deployment.yaml#L324)
